Make full-screen dialog text editable in gallery (#13690)

lwoerdeman opened the pull request.

lwoerdeman committed:
Make full-screen dialog text editable in gallery (#13690)

lwoerdeman commented:
I signed it!

HansMuller commented:
This is a nice idea for improving the demo.

If the text fields are editable they should set the _saveNeeded flag. 

This would be a good opportunity to upgrade the appearance of the demo with filled/underlined text fields. See:
https://github.com/flutter/flutter/blob/master/examples/flutter_gallery/lib/demo/material/text_form_field_demo.dart

lwoerdeman committed:
Add _saveNeeded flag to TextField full screen dialog demo (#13690)

lwoerdeman committed:
Add filled, labeled TextFields. Set dialog title to event name (#13690)

HansMuller commented on examples/flutter_gallery/lib/demo/material/full_screen_dialog_demo.dart:
```diff
@@ -164,19 +165,41 @@ class FullScreenDialogDemoState extends State<FullScreenDialogDemo> {
+              child: new TextField(
+                decoration: const InputDecoration(
+                  labelText: 'Event name',
+                  hintText: 'What is the name of the event?',
```
This text doesn't always fit, on my Nexus 5 it becomes "What is the name of the...". Since the labelText makes the field's purpose pretty clear; best to leave out the hintText.

HansMuller commented on examples/flutter_gallery/lib/demo/material/full_screen_dialog_demo.dart:
```diff
@@ -106,6 +106,7 @@ class FullScreenDialogDemoState extends State<FullScreenDialogDemo> {
   DateTime _toDateTime = new DateTime.now();
   bool _allDayValue = false;
   bool _saveNeeded = false;
+  String _eventName = 'New event';
```
This ends up being a little confusing because the app's title doesn't match the (initially empty) event name text field.

It might be better if the app's title was initially ``"<Event Name TBD>"``, or something better if you can think of it :-). You could use _eventName == null to indicate that a name hadn't been specified yet.

HansMuller commented on examples/flutter_gallery/lib/demo/material/full_screen_dialog_demo.dart:
```diff
@@ -164,19 +165,41 @@ class FullScreenDialogDemoState extends State<FullScreenDialogDemo> {
+                    } else {
+                      _eventName = 'New event';
+                    }
+                    _saveNeeded = true;
```
Only set _saveNeed when the event name isn't empty

HansMuller commented on examples/flutter_gallery/lib/demo/material/full_screen_dialog_demo.dart:
```diff
@@ -164,19 +165,41 @@ class FullScreenDialogDemoState extends State<FullScreenDialogDemo> {
+                ),
+                onChanged: (String value) {
+                  setState(() {
+                    _saveNeeded = true;
```
Only set _saveNeeded when the location is actually set. Need a _location state variable

HansMuller reviewed (commented):
The demo looks much better, thanks. Forgive me for indulging in a little mission creep here, trying to improve the demo just a little bit more.

Hixie commented:
@lwoerdeman any chance you'll have any time to apply the suggested edits? No problem if not, we can take over the patch for you if you want.

lwoerdeman committed:
Remove lengthy hint text, improve save needed logic (#13690)

lwoerdeman commented:
@Hixie @HansMuller I've addressed some of the comments, leaving out the location state field since it wouldn't be used by anything. Sorry for the delay, got caught up with other things.

HansMuller commented:
Thanks for sticking with this!

LGTM. I will land it shortly.
